DQMOffline/Trigger/plugins/METMonitor.cc

```diff
@@ -153,62 +188,101 @@ void METMonitor::bookHistograms(DQMStore::IBooker     & ibooker,
-
+  std::cout<<"debug1"<<std::endl;
   edm::Handle<reco::PFMETCollection> metHandle;
   iEvent.getByToken( metToken_, metHandle );
   reco::PFMET pfmet = metHandle->front();
   if ( ! metSelection_( pfmet ) ) return;
-  
+
+  float ht = 0.0; 
   float met = pfmet.pt();
   float phi = pfmet.phi();
-
+  std::cout<<"debug2"<<std::endl;
   edm::Handle<reco::PFJetCollection> jetHandle;
   iEvent.getByToken( jetToken_, jetHandle );
-  std::vector<reco::PFJet> jets;
+  std::vector<reco::PFJet> jets;   std::cout<<"debug2a"<<std::endl;
+  jets.clear();
   if ( int(jetHandle->size()) < njets_ ) return;
-  for ( auto const & j : *jetHandle ) {
-    if ( jetSelection_( j ) ) jets.push_back(j);
-  }
+  std::cout<<"debug2b"<<std::endl;
+  for ( auto const & j : *jetHandle ) {   std::cout<<"debug2c"<<std::endl;
+    if ( jetSelection_(j) ) {   std::cout<<"debug2d"<<std::endl;
+      jets.push_back(j);   std::cout<<"debug2e"<<std::endl;
+      if ( jetSelection_HT_(j)) ht += j.pt();   std::cout<<"debug2f"<<std::endl;
+    }
+  }   std::cout<<"debug2g"<<std::endl;
   if ( int(jets.size()) < njets_ ) return;
-  
+  std::cout<<"debug3"<<std::endl;
+  float deltaPhi_met_j1= 10.0;
+  float deltaPhi_j1_j2 = 10.0;
+
+  if (int(jets.size()) >= 1) deltaPhi_met_j1 = fabs( deltaPhi( pfmet.phi(),  jets[0].phi() ));
+  if (int(jets.size()) >= 2) deltaPhi_j1_j2 = fabs( deltaPhi( jets[0].phi(),  jets[1].phi() ));
+  std::cout<<"debug4"<<std::endl;
   edm::Handle<reco::GsfElectronCollection> eleHandle;
   iEvent.getByToken( eleToken_, eleHandle );
   std::vector<reco::GsfElectron> electrons;
+  electrons.clear();
   if ( int(eleHandle->size()) < nelectrons_ ) return;
   for ( auto const & e : *eleHandle ) {
     if ( eleSelection_( e ) ) electrons.push_back(e);
   }
   if ( int(electrons.size()) < nelectrons_ ) return;
   
+  edm::Handle<reco::VertexCollection> vtxHandle;
+  iEvent.getByToken(vtxToken_, vtxHandle);
+
+  reco::Vertex vtx;
+  math::XYZPoint pv(0, 0, 0);
+  for (vector<reco::Vertex>::const_iterator v = vtxHandle->begin(); v != vtxHandle->end(); ++v) {
+    bool isFake =  v->isFake() ;
+    
+    if (!isFake) {
+      pv.SetXYZ(v->x(), v->y(), v->z());
+      vtx = *v;
+      break;
+    }
+  }
+
   edm::Handle<reco::MuonCollection> muoHandle;
   iEvent.getByToken( muoToken_, muoHandle );
   if ( int(muoHandle->size()) < nmuons_ ) return;
   std::vector<reco::Muon> muons;
+  muons.clear();
   for ( auto const & m : *muoHandle ) {
-    if ( muoSelection_( m ) ) muons.push_back(m);
+    if ( muoSelection_( m ) && m.isGlobalMuon() && m.isPFMuon() && m.globalTrack()->normalizedChi2() < 10. && m.globalTrack()->hitPattern().numberOfValidMuonHits() > 0 && m.numberOfMatchedStations() > 1 && fabs(m.muonBestTrack()->dxy(pv)) < 0.2 && fabs(m.muonBestTrack()->dz(pv)) < 0.5 && m.innerTrack()->hitPattern().numberOfValidPixelHits() > 0 && m.innerTrack()->hitPattern().trackerLayersWithMeasurement() > 5 ) muons.push_back(m);
```

Comment:
Actually the selection cannot be done in the string parser, it requires the primary vertex.